seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/trino/TrinoTypeMapper.java

```diff
@@ -0,0 +1,119 @@
+
+    // -------------------------string----------------------------
+    private static final String TRINO_CHAR = "CHAR";
+    private static final String TRINO_VARCHAR = "VARCHAR";
+    private static final String TRINO_JSON = "JSON";
+
+    // ------------------------------time-------------------------
+    private static final String TRINO_DATE = "DATE";
+    private static final String TRINO_TIME = "TIME";
+    private static final String TRINO_TIMESTAMP = "TIMESTAMP";
+
+    // ------------------------------blob-------------------------
+    private static final String TRINO_BINARY = "BINARY";
+    private static final String TRINO_VARBINARY = "VARBINARY";
+
+    @SuppressWarnings("checkstyle:MagicNumber")
+    @Override
+    public SeaTunnelDataType<?> mapping(ResultSetMetaData metadata, int colIndex)
+            throws SQLException {
+        String columnType = metadata.getColumnTypeName(colIndex).toUpperCase();
+        // VARCHAR(x)      --->      VARCHAR
+        if (columnType.indexOf("(") > -1) {
+            columnType = columnType.split("\\(")[0];
+        }
+        int precision = metadata.getPrecision(colIndex);
+        int scale = metadata.getScale(colIndex);
+        switch (columnType) {
+            case TRINO_BOOLEAN:
+                return BasicType.BOOLEAN_TYPE;
+            case TRINO_TINYINT:
+                return BasicType.BYTE_TYPE;
+            case TRINO_INTEGER:
+                return BasicType.INT_TYPE;
+            case TRINO_SMALLINT:
+                return BasicType.SHORT_TYPE;
+            case TRINO_BIGINT:
+                return BasicType.LONG_TYPE;
+            case TRINO_DECIMAL:
+                return new DecimalType(precision, scale);
+            case TRINO_REAL:
+                return BasicType.FLOAT_TYPE;
+            case TRINO_DOUBLE:
+                return BasicType.DOUBLE_TYPE;
+            case TRINO_CHAR:
+            case TRINO_VARCHAR:
+            case TRINO_JSON:
+                return BasicType.STRING_TYPE;
+            case TRINO_DATE:
+                return LocalTimeType.LOCAL_DATE_TYPE;
+            case TRINO_TIME:
+                return LocalTimeType.LOCAL_TIME_TYPE;
+            case TRINO_TIMESTAMP:
+                return LocalTimeType.LOCAL_DATE_TIME_TYPE;
+            case TRINO_VARBINARY:
+            case TRINO_BINARY:
+                return PrimitiveByteArrayType.INSTANCE;
+                // Doesn't support yet
+            case TRINO_MAP:
+            case TRINO_ARRAY:
+            case TRINO_ROW:
+            default:
+                final String jdbcColumnName = metadata.getColumnName(colIndex);
+                throw new JdbcConnectorException(
+                        CommonErrorCode.UNSUPPORTED_OPERATION,
+                        String.format(
+                                "Doesn't support TRINO type '%s' on column '%s'  yet.",
+                                columnType, jdbcColumnName));
+        }
```

Comment:
Please create common error exception through https://github.com/apache/seatunnel/blob/392300ea023f3879c84a9b98a61da19a0e31ceb0/seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonError.java#L58

Reply:
did the same way